runtime/runtime-core/common/src/aws/smithy/kotlin/runtime/content/BigDecimal.kt

```diff
@@ -0,0 +1,9 @@
+/*
+ * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+package aws.smithy.kotlin.runtime.content
+
+public expect class BigDecimal(value: String) : Number {
```

Comment:
**correctness/question**: Should this implement `Number`?

Reply:
Certainly up for debate. In my opinion, they should because they are numbers (taxonomically). Kotlin's stdlib [`Number` alias](https://kotlinlang.org/api/latest/jvm/stdlib/kotlin/-number/) automatically adds methods for conversion to other types (e.g., `toInt`, `toLong`, etc.) which seems useful. Plus I believe there are areas in our code where we check if something is a `Number` and act accordingly.

Reply:
I suppose `Number` documents that these methods may truncate. I guess it's fine 🤷